### DISCUSSION TITLE
The github copilot cannot connect to the server

### DISCUSSION THREAD

**Discussion by zym0212** [community/community#8639] [Copilot Conversations] [answered]
![image](https://user-images.githubusercontent.com/93334597/145518563-d595acf4-db5c-4490-9c3c-9731b4eef685.png)
I have tried to reboot the program, and installed the program again, and also changed the Internet configures, but the copilot still not works.

**Comment by zym0212** [comment]
the IDE is visual studio code.

**Reply by zym0212** [reply]
I received activation message months ago and I have been used this for quite a while. One day it just failed.

**Comment by davecheney** [comment]
is it possible that the GFW is blocking your connection to api.github.com?

**Reply by zym0212** [reply]
Thank you for the answer. I noticed that I could not get access to github without a VPN a week ago, and that was when the copilot failed I guess. But I still can't connect to api.github.com with my VPN on.

**Reply by zym0212** [reply]
I added a  DNS of github to the hosts documents, now I can get access to the github without VPN but still cannot use copilot.

**Reply by zym0212** [reply]
I hate GFW. Worst in the globe.

**Comment by zym0212** [comment, marked as answer]
OK, I think now I find what the problem is. The BUAA university switch to a new and cheaper Internet server and they blocked the connection to api.github.com. Now I'm using my hotspot with my mobile phone and it works. It works!!!!

**Comment by incrediblegenius** [comment]
Actually, I cant get access to 'api.github.com' via Chinese mainland university net(with VPNs). Although I just use Copilot by using mobile phone hotspot net F...

**Reply by zym0212** [reply]
Thank you for reply. BUAA university 'upgrade' its network service this month, providing cheaper Internet, but block the access to many sites including github. Even the Internet in my home can connect to github. BUAA is trying to save money without being noticed.

**Comment by billsaysthis** [comment]
Just got my invite to join and while I was able to install the VSC extension and authorize it, when I try to follow the first example in the docs I get the above error and this is in OUTPUT:

```[ERROR] [default] [2022-06-11T00:11:29.391Z] GitHub Copilot could not connect to server. Extension activation failed: "Cancelled"```

No VPN here.

**Comment by glc12125** [comment]
I had the same issue and the following method fixed it. 

1. Find out the IP of the api.github.com on a website like https://www.nslookup.io/domains/api.github.com/webservers/
2. add one line to the end of /etc/hosts: 192.30.255.117 api.github.com
3. Reload the copilot extension
